hw/ip/keymgr/doc/_index.md

```diff
@@ -244,9 +251,57 @@ Based on input from key manager control, this module selects the inputs for each
 
 ![Key Manager KMAC Interface Block Diagram](keymgr_kmac_if_diagram.svg)
 
-### Software Binding
+### Side Load Keys
+
+There are three sideload keys.
+One for AES, one for HMAC, and one for KMAC.
+When a sideload key is generated successfually through the `generate-output-hw` command, the derived data is loaded into key storage registers.
+There is a set of storage registers for each destination.
+
+The KMAC key however is further overloaded as it is the main derivation mechanism for key manager internal stage.
+The KMAC key thus has two possible outputs, one is the sideload key, and the other is internal state key.
+
+When a valid operation is called, the internal state key is sent over the KMAC key.
+During all other times, the sideloaded value is presented.
+The sideload key can be overwritten with another generate command, or cleared with entropy through {{< regref SIDELOAD_CLEAR >}}.
+
+The following diagram illustrates an example when there is no valid key in the KMAC sideload registers and an operation is called.
+During the duration of the operation, the key is valid and shows the internal key state.
```

Comment:
see comment above.
There are now two sets of storage (i'll try to clarify this).
there's the sideload key storage, and the internal key storage.

They both go to keymgr using the same bus, however one does not overwrite the other.
The sideload key is overrwitten by new generate commands, or key manager enable going low.
Let me know think about whether a transition to disable should also wipe.

Reply:
Sorry, I may not understand this correctly.
if I issue 2 kmac gen-hw, the 2nd gen-hw still needs to go through KDF to generate digest data to use as sideload key, right? If yes, we don't have to store 3 HW sideload keys and prevent them from being overwritten by other operation (advance, gen-sw)

Reply:
so let me clarify.
There are 3 versions of the sideload key because each is directed towards an end point (aes , hmac, kmac).
They each get their own so we don't have to deal with any contamination.

The kmac key specifically, is an overloaded function. It has both the sideload key, and the internal key used for derivation (advance / generate).

So the key manager holds onto the internal key and generates.. the generated key is then stored into sideload storage.
When you call advance, then the internal key is used, and the internal key is updated.

Reply:
you're right. I forgot 3 sideload interfaces are separate.
Since sideload key are stored, not sure if it's worth using the stored key without generation step to save some time, when there is a valid key.